### PR TITLE
Tabular: Improved TextPrediction with category features, improved GPU usage logging.

### PR DESCRIPTION
**Pull request by Innixma**
*Issue #, if available:*

*Description of changes:*

- Improved TextPrediction with category features
- Improved GPU usage logging.
- Trying to train TextPrediction Tabular model without GPU will now error instead of hang the process.
- TextPrediction model will now automatically use GPU if available.
- Added new `fillna` parameter to CategoryFeatureGenerator with option 'mode' to fillna to the mode category.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tabular/src/autogluon/tabular/models/text_prediction/text_prediction_v1_model.py [hunk 1]
```diff
@@ -62,8 +65,21 @@ def __init__(self, **kwargs):
...
         super().__init__(**kwargs)
         self._label_column_name = None
-        self._numeric_columns = None
-        self._cat_columns = None
+        self._feature_generator = None
+
+    def _preprocess(self, X, fit=False, **kwargs):
+        from ...features.feature_metadata import R_CATEGORY
+        if fit:
+            from ...features.generators import BulkFeatureGenerator, CategoryFeatureGenerator, IdentityFeatureGenerator
```
Comment (sxjscience) [inline tabular/src/autogluon/tabular/models/text_prediction/text_prediction_v1_model.py:73]: One minor question. I feel that these feature generators are super useful. Will we later move that to core?

Comment (Innixma) [reply tabular/src/autogluon/tabular/models/text_prediction/text_prediction_v1_model.py:73]: The plan is to actually move them to a new module called `autogluon.features`, which will only depend on `autogluon.core`. This is the next major effort planned after API refactor. The hope is for `autogluon.features` to be in v0.1 release.

**Review by sxjscience — APPROVED** [review]
LGTM

**Comment by szha** [comment]
Job PR-867-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-867/1/index.html

**Comment by szha** [comment]
Job PR-867-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-867/2/index.html

**Pull request merged**